images/egress/dns-proxy/egress_dns_proxy_test.go

```diff
@@ -0,0 +1,255 @@
+		{
+			src:  "1.2.3.4",
+			dest: "80 11.12.13.14",
+			frontends: []string{`
+frontend fe1
+    bind 1.2.3.4:80
+    default_backend be1`},
+			backends: []string{`
+backend be1
+    server dest1 11.12.13.14:80 check`},
+		},
+		// Multiple destination IPs
+		{
+			src:  "1.2.3.4",
+			dest: "80 11.12.13.14\n8080 21.22.23.24 100",
+			frontends: []string{`
+frontend fe1
+    bind 1.2.3.4:80
+    default_backend be1`, `
+frontend fe2
+    bind 1.2.3.4:8080
+    default_backend be2`},
+			backends: []string{`
+backend be1
+    server dest1 11.12.13.14:80 check`, `
+backend be2
+    server dest2 21.22.23.24:100 check`},
+		},
+		// Single destination domain name
+		{
+			src:  "1.2.3.4",
+			dest: "80 example.com",
+			frontends: []string{`
+frontend fe1
+    bind 1.2.3.4:80
+    default_backend be1`},
+			backends: []string{`
+backend be1
+    server dest1 example.com:80 check resolvers dns-resolver`},
+		},
+		// Multiple destination domain names
+		{
+			src:  "1.2.3.4",
+			dest: "80 example.com\n8080 foo.com 100",
+			frontends: []string{`
+frontend fe1
+    bind 1.2.3.4:80
+    default_backend be1`, `
+frontend fe2
+    bind 1.2.3.4:8080
+    default_backend be2`},
+			backends: []string{`
+backend be1
+    server dest1 example.com:80 check resolvers dns-resolver`, `
+backend be2
+    server dest2 foo.com:100 check resolvers dns-resolver`},
+		},
+		// Destination IP and destination domain name
+		{
+			src:  "1.2.3.4",
+			dest: "80 11.12.13.14\n8080 example.com 100",
+			frontends: []string{`
+frontend fe1
+    bind 1.2.3.4:80
+    default_backend be1`, `
+frontend fe2
+    bind 1.2.3.4:8080
+    default_backend be2`},
+			backends: []string{`
+backend be1
+    server dest1 11.12.13.14:80 check`, `
+backend be2
+    server dest2 example.com:100 check resolvers dns-resolver`},
+		},
+		// Destination with comments and blank lines
+		{
+			src: "1.2.3.4",
+			dest: `
+# My DNS proxy egress router rules
+
+# Port 80 forwards to 11.12.13.14
+80 11.12.13.14
+
+# Port 8080 forwards to port 100 on example.com
+8080 example.com 100
+
+# Skip this rule for now
+# 9000 foo.com 200
+
+# End
+`,
+			frontends: []string{`
+frontend fe1
+    bind 1.2.3.4:80
+    default_backend be1`, `
+frontend fe2
+    bind 1.2.3.4:8080
+    default_backend be2`},
+			backends: []string{`
+backend be1
+    server dest1 11.12.13.14:80 check`, `
+backend be2
+    server dest2 example.com:100 check resolvers dns-resolver`},
+		},
+	}
+
+	for n, test := range tests {
+		cmd := exec.Command("./egress-dns-proxy.sh")
+		cmd.Env = []string{
+			fmt.Sprintf("EGRESS_DESTINATION=%s", test.dest),
+			fmt.Sprintf("EGRESS_SOURCE=%s", test.src),
+			fmt.Sprintf("EGRESS_DNS_PROXY_MODE=unit-test"),
+		}
+		outBytes, err := cmd.CombinedOutput()
+		if err != nil {
+			t.Fatalf("test %d unexpected error %v, output: %q", n+1, err, string(outBytes))
+		}
+		out := string(outBytes)
+		for _, frontend := range test.frontends {
+			if !strings.Contains(out, frontend) {
```

Comment:
Did you fix this? Basically I was suggesting you should parse the output into static config, resolvers, and frontends sections, so you could then just check for equality rather than just "contains". (This might be easier if you had multiple testing modes; eg, one that just runs `generate_dns_resolvers` and nothing else, and one that just runs `generate_haproxy_frontends_backends` and nothing else.)